In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/iris/Iris.csv
/kaggle/input/iris/database.sqlite


In [2]:
data = pd.read_csv('/kaggle/input/iris/Iris.csv')

## Basic Statistics

### Analysing the data to understand insights


Variables and cases


Characteristics of football players --> cases

1. Body weight 
2. hair colour
3. age
4. goal


Characteriestics of Team --> cases
1. country
2. color
3. Goals



##### Variable should vary!!

Categorial --> Ordinal & Nominal

Quantitative --> Interval & Ratio

Quantitative --> Discrete and Continous

Data matrix as the summaries of data..


In [3]:
data['Species']  # Categorical

0         Iris-setosa
1         Iris-setosa
2         Iris-setosa
3         Iris-setosa
4         Iris-setosa
            ...      
145    Iris-virginica
146    Iris-virginica
147    Iris-virginica
148    Iris-virginica
149    Iris-virginica
Name: Species, Length: 150, dtype: object

In [4]:
data['PetalLengthCm'] # Numerical

0      1.4
1      1.4
2      1.3
3      1.5
4      1.4
      ... 
145    5.2
146    5.0
147    5.2
148    5.4
149    5.1
Name: PetalLengthCm, Length: 150, dtype: float64

##### Frequency tables
It shows how the value are distributed




In [5]:

df = data['Species'].value_counts() 
print(df) 


Species
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: count, dtype: int64


In [6]:

freq_table = pd.crosstab(data['Species'], 'no_of_species') 

freq_table 


col_0            no_of_species
Species                       
Iris-setosa                 50
Iris-versicolor             50
Iris-virginica              50

Pie chart - Nominal or Ordinal


Bar chart -  Nominal or Ordinal


Dot plot - Quantitative


Histogram - Interval or Ratio


##### Measure of Central dentency

1. Median,-- Quantitative with Outliers or Skewed

2. Mode, -- Categorical

3. Mean, -- Quantitative with Bell curve

##### Measure of dispersion
 
Range, -- Maximum - Minimum

IQR, -- Q3 - Q1      Outlier = Q1 or Q3 (+ or _) 1.5 IQR

###### takes all values into account

Variance, --     (x - x~ ) 2 / n-1   --> larger the variance larger the variablity

Standard deviation, --  Sq. Root of ((x - x~ ) 2 / n-1)  --> often used measure of dispersion

##### Z Score - gives idea if an event is common or exceptional

###### Z = x - x~  / SD

1. -ve z score reprents below the mean
2. +ve z score Above the mean

Normal distribution - Z scores are 68% -1 or +1...   95% -2 to +2  and  99% -3 to +3 

Skewed Right - large Positive Z scores are common

Skewed Left - large Negative Z scores are common


Z score of -1 to 1 = 75%

Z score of -2 to +2 = 90%



##### Standardization:

Replacing the orihinal scores with SD from the mean..